tests/unit/python/foglamp/services/core/scheduler/test_scheduler.py

```diff
@@ -0,0 +1,1184 @@
+        scheduler._schedule_first_task(sch, current_time)
+        time_after_call = sch_execution.next_start_time
+
+        # THEN
+        assert time_after_call > time.mktime(curr_time.timetuple())
+
+    @pytest.mark.asyncio
+    async def test__get_process_scripts(self, mocker):
+        # GIVEN
+        scheduler = Scheduler()
+        scheduler._storage = Storage(core_management_host=None, core_management_port=None)
+
+        # WHEN
+        await scheduler._get_process_scripts()
+
+        # THEN
+        assert 14 == len(scheduler._process_scripts)
+
+    @pytest.mark.asyncio
+    async def test__get_schedules(self, mocker):
+        # GIVEN
+        scheduler = Scheduler()
+        scheduler._storage = Storage(core_management_host=None, core_management_port=None)
+        mocker.patch.object(scheduler, '_schedule_first_task')
+
+        # WHEN
+        await scheduler._get_schedules()
+
+        # THEN
+        assert 14 == len(scheduler._schedules)
+
+    @pytest.mark.asyncio
+    async def test__read_storage(self, mocker):
+        # GIVEN
+        scheduler = Scheduler()
+        scheduler._storage = Storage(core_management_host=None, core_management_port=None)
+        mocker.patch.object(scheduler, '_schedule_first_task')
+
+        # WHEN
+        await scheduler._read_storage()
+
+        # THEN
+        assert 14 == len(scheduler._process_scripts)
+        assert 14 == len(scheduler._schedules)
+
+    @pytest.mark.asyncio
+    @pytest.mark.skip("_mark_tasks_interrupted() not implemented in main Scheduler class.")
+    async def test__mark_tasks_interrupted(self, mocker):
+        pass
+
+    @pytest.mark.asyncio
+    @pytest.mark.skip("This unit test will be created after merger of Configuration manager unit tests PR.")
+    async def test__read_config(self, mocker):
+        pass
+
+    @pytest.mark.asyncio
+    async def test_start(self, mocker):
+        # TODO: Mandatory - Add negative tests for full code coverage
+        # GIVEN
+        scheduler = Scheduler()
+        scheduler._storage = Storage(core_management_host=None, core_management_port=None)
+        current_time = time.time()
+        curr_time = datetime.datetime.fromtimestamp(current_time)
+        mocker.patch.object(scheduler, '_schedule_first_task')
+        mocker.patch.object(scheduler, '_scheduler_loop', return_value=asyncio.sleep(.1))
+        mocker.patch.multiple(scheduler, _core_management_port=9999,
+                              _core_management_host="0.0.0.0",
+                              current_time=current_time - 3600)
+
+        # TODO: Remove after implementation of above test test__read_config()
+        mocker.patch.object(scheduler, '_read_config', return_value=asyncio.sleep(.1))
+
+        # Confirm that Scheduler is not running
+        assert scheduler._ready is False
+
+        # WHEN
+        await scheduler.start()
+
+        # THEN
+        assert scheduler._ready is True
+        assert 14 == len(scheduler._process_scripts)
```

Comment:
Use of constants like this feels fragile as this relates to structures defined below there is a tendency for this to get broken if/when new tests modify those structures below.